src/main.cpp

```diff
@@ -919,19 +931,54 @@ bool ContextualCheckTransaction(
                 }
 
                 // SaplingNotePlaintext::decrypt() checks note commitment validity.
-                if (!SaplingNotePlaintext::decrypt(
+                auto notePlaintext = SaplingNotePlaintext::decrypt(
                     output.encCiphertext,
                     output.ephemeralKey,
                     outPlaintext->esk,
                     outPlaintext->pk_d,
-                    output.cmu)
-                ) {
+                    output.cmu);
+
+                if (!notePlaintext) {
                     return state.DoS(
                         DOS_LEVEL_BLOCK,
                         error("CheckTransaction(): coinbase output description has invalid encCiphertext"),
                         REJECT_INVALID,
                         "bad-cb-output-desc-invalid-encct");
                 }
+
+                // Detect any ZIP 207 shielded funding streams.
+                if (canopyActive && tx.IsCoinBase()) {
```

Comment:
`tx.IsCoinBase()` is redundant here as this has already been established higher up in the `if` nesting.